### PR TITLE
 Remove hardcoded root  Network / Perf-Network

### PR DESCRIPTION
**Pull request by asofron**
Affected test cases : 

 **Testscripts/Windows/NET-IP-Injection.ps1**
 1 TC NET-IP-INJECTION
 **Testscripts/Windows/NET-TCP-Corruption.ps1**
 1 TC NET-TCP-CORRUPTION
 **Testscripts/Windows/NET-Test-Private-Network.ps1**
 2 TC NET-CHANGE-PRIVATE-NIC NET-PRIVATE
 **Testscripts/Windows/NET-Test-Vxlan.ps1**
  1 TC NET-VXLAN
 **Testscripts/Windows/NET-VLAN-Tagging.ps1**
  1 TC NET-VLAN-TAGGING
 **Testscripts\Windows\NET-VLAN-Trunking.ps1**
 1 TC NET-VLAN-TRUNKING
**Testscripts/Windows/VERIFY-ENABLE-DISABLE-SRIOV-ON-EXISTING-VM.ps1**
1 TC VERIFY-ENABLE-DISABLE-SRIOV-ON-EXISTING-VM
**Testscripts/Windows/SETUP-Inject-IP-Constants.ps1** 
 10 TC
 NET-INTERNAL
 NET-EXTERNAL
 NET-CHANGE-INTERNAL-NIC
 NET-JUMBO-FRAMES
 NET-COPY-LARGE-FILE
 NET-COPY-FILE-DIFFERENT-MTU
 NET-COPY-BOTH-WAYS
 NET-VXLAN
 NET-MULTICAST
 ETHTOOL-CHECK-STATISTICS
 **Testscripts/Windows/SETUP-NET-Configure-Dependency-VM.ps1** 
10TC
 NET-INTERNAL
 NET-EXTERNAL
 NET-CHANGE-INTERNAL-NIC
 NET-JUMBO-FRAMES
 NET-COPY-LARGE-FILE
 NET-COPY-FILE-DIFFERENT-MTU
 NET-COPY-BOTH-WAYS
 NET-VXLAN
 NET-MULTICAST
 ETHTOOL-CHECK-STATISTICS
 **Testscripts/Windows/SETUP-PERF-Add-NIC.ps1**
 12 TC
 PERF-NETWORK-TCP-THROUGHPUT-MULTICONNECTION-NTTTCP-Synthetic
 PERF-NETWORK-TCP-THROUGHPUT-MULTICONNECTION-NTTTCP-SRIOV
 PERF-NETWORK-UDP-1K-THROUGHPUT-MULTICONNECTION-NTTTCP-Synthetic
 PERF-NETWORK-TCP-LATENCY-MULTICONNECTION-Synthetic
 PERF-NETWORK-TCP-LATENCY-MULTICONNECTION-SRIOV
 PERF-NETWORK-UDP-THROUGHPUT-MULTICONNECTION-Synthetic
 PERF-NETWORK-UDP-THROUGHPUT-MULTICONNECTION-SRIOV
 PERF-NETWORK-TCP-SINGLE-CONNECTION-THROUGHPUT-SYNTHETIC
 PERF-NETWORK-TCP-SINGLE-CONNECTION-THROUGHPUT-SRIOV
 PERF-NETWORK-TCP-NETPERF-SINGLE-PPS-Synthetic
 PERF-NETWORK-TCP-NETPERF-SINGLE-PPS-SRIOV
 PERF-NETWORK-TCP-NETPERF-MAX-PPS-Synthetic
 PERF-NETWORK-TCP-NETPERF-MAX-PPS-SRIOV
 **Testscripts/Windows/SETUP-SR-IOV-Configure.ps1**
 24TC
STRESSTEST-WEB-PARALLEL-ACCESS
 VERIFY-SRIOV-LSPCI
 SRIOV-VERIFY-VF-BASIC-CONNECTION
 SRIOV-VERIFY-VF-BASIC-CONNECTION-ONE-VCPU
 SRIOV-VERIFY-VF-BASIC-CONNECTION-MAX-VCPU
 SRIOV-VERIFY-VF-MULTIPLE-CONNECTION
 SRIOV-VERIFY-VF-MAX-CONNECTION
 SRIOV-VERIFY-VF-MAX-CONNECTION-MAX-VCPU
 SRIOV-IPERF-STRESS
 SRIOV-DISABLEVF-ON-GUEST
 SRIOV-RELOAD-MODULE
 SRIOV-DISABLEVF-PING
 SRIOV-DISABLEVF-IPERF
 SRIOV-DETACH-NIC
 SRIOV-DISABLEVF-ONHOST
 SRIOV-DISABLE-NIC
 SRIOV-DISABLE-VMQ
 SRIOV-CHANGE-RSS
 SRIOV-MEASURE-VF-FAILBACK
 SRIOV-REBOOT-VM
 SRIOV-MOVE-VHD
 SRIOV-STRESS-SAVE
 SRIOV-STRESS-PAUSE
 SRIOV-MULTICAST
 SRIOV-BROADCAST

**Do not merge until test results are posted**

### CODE REVIEW

File: Testscripts/Linux/SR-IOV-Utils.sh [hunk 1]
```diff
@@ -133,7 +133,7 @@ ConfigureVF()
     if [ $vfCount -eq 0 ]; then
         LogErr "No VFs are present in the Guest VM!"
         SetTestStateFailed
-        exit 1
+        exit 0
```
Comment (LiliDeng) [inline Testscripts/Linux/SR-IOV-Utils.sh]: why change this?

File: Testscripts/Linux/enablePasswordLessRoot.sh [hunk 2]
```diff
@@ -6,21 +6,39 @@
 # In multi VM cluster. Execute this script in one VM. It will create a sshFix.tar
 # Copy this sshFix.tar to other VMs (/root) in your cluster and execute same script. It will extract previously created keys.
 # This way, all VMs will have same public and private keys in .ssh folder.
+set -xe
+
+if [[ $1 != '' ]]; then
+Custom_Path=$1
```
Comment (LiliDeng) [inline Testscripts/Linux/enablePasswordLessRoot.sh]: This file has indentation issues

File: Testscripts/Windows/NET-Test-Private-Network.ps1 [hunk 3]
```diff
@@ -27,7 +27,7 @@ function Main {
     $switchNic = $null
     $bootproto = "static"
     $currentDir = "$pwd\"
-    $guestUsername = "root"
+    $VMUserName = "$VMUserName"
```
Comment (LiliDeng) [inline Testscripts/Windows/NET-Test-Private-Network.ps1]: remove

File: Testscripts/Windows/SRIOV-DISABLE-VMQ.ps1 [hunk 4]
```diff
@@ -89,4 +89,4 @@ function Main {
 }
 
 Main -VMName $AllVMData.RoleName -hvServer $xmlConfig.config.Hyperv.Hosts.ChildNodes[0].ServerName `
-    -VMPort $AllVMData.SSHPort -VMPassword $password
+    -VMPort $AllVMData.SSHPort -VmUsername $user  -VMPassword $password
```
Comment (LiliDeng) [inline Testscripts/Windows/SRIOV-DISABLE-VMQ.ps1]: extra 1 space between  -VmUsername $user  -VMPassword $password?

File: Testscripts/Windows/SRIOV-DISABLEVF-IPERF.ps1 [hunk 5]
```diff
@@ -96,4 +96,4 @@ function Main {
 }
 
 Main -VMName $AllVMData.RoleName -hvServer $xmlConfig.config.Hyperv.Hosts.ChildNodes[0].ServerName `
-    -VMPort $AllVMData.SSHPort -VMPassword $password
+    -VMPort $AllVMData.SSHPort  -VMUsername $user  -VMPassword $password
```
Comment (LiliDeng) [inline Testscripts/Windows/SRIOV-DISABLEVF-IPERF.ps1]: ditto

File: Testscripts/Windows/SRIOV-REBOOT-VM.ps1 [hunk 6]
```diff
@@ -16,54 +16,55 @@ param ([string] $TestParams)
...
 function Main {
     param (
+        $VMUsername,
         $VMName,
         $HvServer,
         $VMPort,
         $VMPassword
     )
-    $VMRootUser = "root"
+    $VMUsername = "$user"
```
Comment (LiliDeng) [inline Testscripts/Windows/SRIOV-REBOOT-VM.ps1]: remove

File: Testscripts/Windows/SRIOV-RUN-COMMON-TEST.ps1 [hunk 7]
```diff
@@ -83,29 +82,29 @@ function Main {
                 Write-LogInfo "Setting SSH keys for both VMs"
                 Copy-RemoteFiles -uploadTo $publicIp -port $vmPort -files `
                     ".\Testscripts\Linux\enablePasswordLessRoot.sh,.\Testscripts\Linux\utils.sh,.\Testscripts\Linux\SR-IOV-Utils.sh" `
-                    -username $vmRootUser -password $password -upload
+                    -username $user -password $password -upload
```
Comment (LiliDeng) [inline Testscripts/Windows/SRIOV-RUN-COMMON-TEST.ps1]: why not keep consistent, use $VMUsername parameter to pass in?

File: Testscripts/Windows/SRIOV-RUN-COMMON-TEST.ps1 [hunk 8]
```diff
@@ -83,29 +82,29 @@ function Main {
...
                 Copy-RemoteFiles -uploadTo $publicIp -port $dependencyVmData.SSHPort -files `
                     ".\Testscripts\Linux\enablePasswordLessRoot.sh,.\Testscripts\Linux\utils.sh,.\Testscripts\Linux\SR-IOV-Utils.sh" `
-                    -username $vmRootUser -password $password -upload
-                Run-LinuxCmd -ip $publicIp -port $vmPort -username $vmRootUser -password `
+                    -username $user -password $password -upload
+                Run-LinuxCmd -ip $publicIp -port $vmPort -username $user -password `
                     $password -command "chmod +x ~/*.sh"
```
Comment (LiliDeng) [inline Testscripts/Windows/SRIOV-RUN-COMMON-TEST.ps1]: Files will be copied into /root folder or user folder?

**Comment by LiliDeng** [comment]
I random pick up a case VERIFY-ENABLE-DISABLE-SRIOV-ON-EXISTING-VM, it abort

**Comment by asofron** [comment]
NET :

Hyper-v:

 VHD Under Test        : Centos_7.5_x64.vhdx
 Total Test Cases      : 4 (4 Pass, 0 Fail, 0 Abort)
 Total Time (dd:hh:mm) : 0:0:33
 XML File              : TestConfiguration
 
    ID TestCaseName                                                 TestResult TestDuration(in minutes) 
 ------------------------------------------------------------------------------------------------------
     1 NET-IP-INJECTION                                                   PASS                 13.5 
     2 NET-TCP-CORRUPTION                                                 PASS                 4.49 
     3 NET-VLAN-TAGGING                                                   PASS                 8.08 
     4 NET-VLAN-TRUNKING                                                  PASS                 7.23 

 VHD Under Test        : Centos_7.5_x64.vhdx
 Total Test Cases      : 10 (9 Pass, 1 Fail, 0 Abort)
 Total Time (dd:hh:mm) : 0:1:43
 XML File              : TestConfiguration

    ID TestCaseName                                                 TestResult TestDuration(in minutes) 
 ------------------------------------------------------------------------------------------------------
     1 NET-EXTERNAL                                                       PASS                 4.94 
     2 NET-INTERNAL                                                       PASS                  5.2 
     3 NET-CHANGE-INTERNAL-NIC                                            PASS                 4.79 
     4 NET-JUMBO-FRAMES                                                   PASS                 9.93 
     5 NET-COPY-LARGE-FILE                                                PASS                  8.3 
     6 NET-COPY-FILE-DIFFERENT-MTU                                        PASS                 8.36 
     7 NET-COPY-BOTH-WAYS                                                 PASS                26.62 
     8 NET-VXLAN                                                          PASS                 13.3 
     9 NET-MULTICAST                                                      PASS                 9.04 
    10 ETHTOOL-CHECK-STATISTICS                                           FAIL                13.36 

Tests that fail because of external causes:
ETHTOOL-CHECK-STATISTICS:
test on tx_send_full param failed.

**Comment by asofron** [comment]
**NET-PERF :**

**Azure:**

ARM Image Under Test  : OpenLogic : CentOS : 7.5 : latest
Total Test Cases      : 14 (14 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:2:40
XML File              : TestConfiguration

   ID TestCaseName                                                 TestResult TestDuration(in minutes)
------------------------------------------------------------------------------------------------------
    1 PERF-NETWORK-TCP-THROUGHPUT-MULTICONNECTION-NTTTCP-Synthetic                 PASS                 8.64
	Connections=1 : throughput=3.49Gbps cyclePerBytet=1.46 Avg_TCP_lat=235.760
    2 PERF-NETWORK-TCP-THROUGHPUT-MULTICONNECTION-NTTTCP-SRIOV                 PASS                 8.22
	Connections=1 : throughput=13.78Gbps cyclePerBytet=1.19 Avg_TCP_lat=100.386
    3 PERF-NETWORK-UDP-1K-THROUGHPUT-MULTICONNECTION-NTTTCP-Synthetic                 PASS                 8.26
	Connections=1 : tx_throughput=1.95Gbps rx_throughput=0.66Gbps datagram_loss=66.15
    4 PERF-NETWORK-UDP-1K-THROUGHPUT-MULTICONNECTION-NTTTCP-SRIOV             PASS        8.22
	Connections=1 : tx_throughput=1.20Gbps rx_throughput=1.12Gbps datagram_loss=6.67
    5 PERF-NETWORK-TCP-LATENCY-MULTICONNECTION-Synthetic                 PASS                14.49
    6 PERF-NETWORK-TCP-LATENCY-MULTICONNECTION-SRIOV                     PASS                 9.72
    7 PERF-NETWORK-UDP-THROUGHPUT-MULTICONNECTION-Synthetic                 PASS                17.52
    8 PERF-NETWORK-UDP-THROUGHPUT-MULTICONNECTION-SRIOV                  PASS                18.11
    9 PERF-NETWORK-TCP-SINGLE-CONNECTION-THROUGHPUT-SYNTHETIC                 PASS             7.84
   10 PERF-NETWORK-TCP-SINGLE-CONNECTION-THROUGHPUT-SRIOV                 PASS                 7.36
   11 PERF-NETWORK-TCP-NETPERF-SINGLE-PPS-Synthetic                      PASS                13.52
	Rx Average PPS : 6347
	Rx Minimum PPS : 5411
	Rx Maximum PPS : 6903
   12 PERF-NETWORK-TCP-NETPERF-SINGLE-PPS-SRIOV                          PASS                12.34
	Rx Average PPS : 18526
	Rx Minimum PPS : 12565
	Rx Maximum PPS : 20339
   13 PERF-NETWORK-TCP-NETPERF-MAX-PPS-Synthetic                         PASS                13.38
	Rx Average PPS : 155540
	Rx Minimum PPS : 8201
	Rx Maximum PPS : 160213
   14 PERF-NETWORK-TCP-NETPERF-MAX-PPS-SRIOV                             PASS                13.24
	Rx Average PPS : 320765
	Rx Minimum PPS : 19132
	Rx Maximum PPS : 345115

File: Testscripts/Windows/SRIOV-DISABLE-VMQ.ps1 (hunk 4 above)
Comment (asofron) [reply Testscripts/Windows/SRIOV-DISABLE-VMQ.ps1]: fixed

File: Testscripts/Linux/enablePasswordLessRoot.sh (hunk 2 above)
Comment (asofron) [reply Testscripts/Linux/enablePasswordLessRoot.sh]: fixed

File: Testscripts/Windows/NET-Test-Private-Network.ps1 (hunk 3 above)
Comment (asofron) [reply Testscripts/Windows/NET-Test-Private-Network.ps1]: fixed

File: Testscripts/Windows/SRIOV-DISABLEVF-IPERF.ps1 (hunk 5 above)
Comment (asofron) [reply Testscripts/Windows/SRIOV-DISABLEVF-IPERF.ps1]: Fixed

File: Testscripts/Windows/SRIOV-REBOOT-VM.ps1 (hunk 6 above)
Comment (asofron) [reply Testscripts/Windows/SRIOV-REBOOT-VM.ps1]: Fixed

File: Testscripts/Windows/SRIOV-RUN-COMMON-TEST.ps1 (hunk 7 above)
Comment (asofron) [reply Testscripts/Windows/SRIOV-RUN-COMMON-TEST.ps1]: Fixed

File: Testscripts/Windows/SRIOV-RUN-COMMON-TEST.ps1 (hunk 8 above)
Comment (asofron) [reply Testscripts/Windows/SRIOV-RUN-COMMON-TEST.ps1]: user folder

**Pull request merged**